packages/react-components/react-tree/stories/Tree/UseHeadlessFlatTree.stories.tsx

```diff
@@ -0,0 +1,148 @@
+//         content: <>level 2, item 2</>,
+//       },
+//       {
+//         value: '1-3',
+//         content: <>level 2, item 3</>,
+//       },
+//     ],
+//   },
+//   {
+//     value: '2',
+//     content: <>level 1, item 2</>,
+//     subtree: [
+//       {
+//         value: '2-1',
+//         content: <>level 2, item 1</>,
+//         subtree: [
+//           {
+//             value: '2-1-1',
+//             content: <>level 3, item 1</>,
+//             subtree: [
+//               {
+//                 value: '2-1-1-1',
+//                 content: <>level 4, item 1</>,
+//               },
+//             ],
+//           },
+//         ],
+//       },
+//     ],
+//   },
+// ];
+
+// const flatTreeItems = flattenTree_unstable<Item>(nestedTreeItems);
+
+const ActionsExample = () => (
+  <>
+    <Button aria-label="Edit" appearance="subtle" icon={<Edit20Regular />} />
+    <Menu>
+      <MenuTrigger disableButtonEnhancement>
+        <Button aria-label="More options" appearance="subtle" icon={<MoreHorizontal20Regular />} />
+      </MenuTrigger>
+
+      <MenuPopover>
+        <MenuList>
+          <MenuItem>New</MenuItem>
+          <MenuItem>New Window</MenuItem>
+          <MenuItem disabled>Open File</MenuItem>
+          <MenuItem>Open Folder</MenuItem>
+        </MenuList>
+      </MenuPopover>
+    </Menu>
+  </>
+);
+
+export const UseHeadlessFlatTree = () => {
+  const flatTree = useHeadlessFlatTree_unstable(flatTreeItems);
+  const focusTargetAttribute = useRestoreFocusTarget();
+
+  return (
+    <FlatTree {...flatTree.getTreeProps()} aria-label="Flat Tree">
+      {Array.from(flatTree.items(), flatTreeItem => {
+        const { content, ...treeItemProps } = flatTreeItem.getTreeItemProps();
+        return (
+          <Menu key={flatTreeItem.value} positioning="below-end" openOnContext>
+            <MenuTrigger disableButtonEnhancement>
+              <TreeItem aria-description="has actions" {...focusTargetAttribute} {...treeItemProps}>
+                <TreeItemLayout actions={<ActionsExample />}>{content}</TreeItemLayout>
+              </TreeItem>
+            </MenuTrigger>
+            <MenuPopover>
+              <MenuList>
+                <MenuItem>Edit</MenuItem>
+                <MenuItem>New</MenuItem>
+                <MenuItem>New Window</MenuItem>
+                <MenuItem disabled>Open File</MenuItem>
+                <MenuItem>Open Folder</MenuItem>
+              </MenuList>
+            </MenuPopover>
+          </Menu>
+        );
+      })}
+    </FlatTree>
+  );
+};
+
+UseHeadlessFlatTree.parameters = {
+  docs: {
+    description: {
+      story: `
+The \`useHeadlessFlatTree\` hook provides all the properties and all the methods required to ensure a proper functioning of a flat tree. It's arguments are:
+
+1. a list of items to be mapped into \`TreeItem\`
+2. an option object that would alow you to control open and checked state
+   * \`openItems\` and \`defaultOpenItems\` (controlling open state)
```

Comment:
nit: if you mention controlling state, it would be nice to mention `onOpenChange`